### DISCUSSION TITLE
Unable to fill a PDF

### DISCUSSION THREAD

**Discussion by kharabela** [pdfcpu/pdfcpu#674] [Q&A]
package main

import (
	"bufio"
	"fmt"
	"os"
	"path/filepath"
	pdfcpu "github.com/pdfcpu/pdfcpu/pkg/api"
)
func main() {

//Print all merge fields in the acro form
InFile, err := os.Open("pdffill/english.pdf")
	if err != nil {
		fmt.Println(err)
	}
	ss, err := pdfcpu.ListFormFields(InFile, pdfcpu.LoadConfiguration())
	for _, s := range ss {
		fmt.Println(s)
	}

//Merge acroform with JSON data
err1 := pdfcpu.FillFormFile("pdffill/english.pdf", "pdffill/english.json", "pdffill/tmp.pdf", pdfcpu.LoadConfiguration())
}
//I am using provided english.pdf and it does print me list of merge fields
//However the output generated is only 0 bytes and doesn't open up

**Comment by hhrutter** [comment]
Hello!

What do you mean by "merge fields"?
You are listing form fields, then you are filling them using JSON - merging is something completely different.
The fill command only cares about fields with provided values in your JSON inFile or a different locking status.

Are you getting any error, any stack trace?
How does your JSON look like?

Are you filling anything at all?

**Comment by hhrutter** [comment]
Closed due to inactivity.
